File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        char max = '\0';
        int n = num.length();
        for(int i = 0;i<=n-3;i++){
            if (num.charAt(i) == num.charAt(i + 1) && num.charAt(i) == num.charAt(i + 2)) {
                max = (char) Math.max(max, num.charAt(i));
            }
        }
        
        return max == '\0' ? "" : new String(new char[]{max,max,max});
    }
}

File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        int n = num.length();
        char c = '\0';
        for(int i = 0;i<=n-3;i++){
            if(num[i] == num[i+1] && num[i] == num[i+2]){
                c = max(c,num[i]);
            }
        }
        
        return c == '\0' ? "" : string(3,c);
    }
};